jujugui/static/gui/src/app/app.js

```diff
@@ -798,22 +798,36 @@ YUI.add('juju-gui', function(Y) {
       element.
 
       @method _renderAddedServices
-      @param {Array} services Array of service models.
+      @param {String} hoveredId An id for a service.
     */
-    _renderAddedServices: function(services) {
+    _renderAddedServices: function(hoveredId) {
       var utils = views.utils;
       var db = this.db;
+      var topo = this.views.environment.instance.topo;
+      var ServiceModule = topo.modules.ServiceModule;
+      // Set up a handler for syncing the service token hover. This needs to be
+      // attached only when the component is visible.
+      if (this.hoverService) {
+        this.hoverService.detach();
+      }
+      this.hoverService = topo.on('hoverService', function(service) {
```

Comment:
Is there a reason why this is inside here? Because this method is called frequently you have to detach and attach the event all the time (as you already know) I feel like this would make more sense in the initializer?